File: binary-tree-inorder-traversal/binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer>res = new ArrayList();
    public List<Integer> inorderTraversal(TreeNode root) {
        //LNR
        /*
        List<Integer> result = new ArrayList();
        Stack<TreeNode> st = new Stack();
        TreeNode curr = root;
        if (curr == null)
            return result;
        while (curr != null || !st.isEmpty()) {
            while(curr != null) {
                st.push(curr);
                curr = curr.left;
            }
            curr = st.pop();
            result.add(curr.val);
            curr = curr.right;
        }
        return result; */
        if( root == null) return res;
        inorderTraversal(root.left);
        res.add(root.val);
        inorderTraversal(root.right);
        return res;
        
    }
}

File: binary-tree-inorder-traversal/binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if not root:
            return []
        stack = []
        res = []
        stack.append(root)
        #print(stack[0].val)
        while stack:
            while root != None and root.left != None:
                print('entered the left subtree')
                stack.append(root.left)
                root = root.left
            temp = stack.pop()
            
            res.append(temp.val)
            print(res)
            if temp != None and temp.right != None:
                
                root = temp.right
                stack.append(root)
        return res